src/main/java/com/ably/kafka/connect/mapping/DefaultMessageSinkMapping.java

```diff
@@ -35,5 +40,20 @@ public Message getMessage(SinkRecord record) {
         return message;
     }
 
+    private Message messageFromRecord(SinkRecord record) {
+        final String messageName = configValueEvaluator.evaluate(record, sinkConnectorConfig.getString(MESSAGE_CONFIG));
+
+        if (record.valueSchema() == null) {
+            return new Message(messageName, record.value());
+        }
 
+        final Schema valueSchema = record.valueSchema();
+        if (valueSchema.type() == Schema.Type.STRUCT) {
+            final String jsonString = StructToJsonConverter.toJsonString((Struct) record.value(), gson);
+            final Message message = new Message(messageName, jsonString);
+            message.encoding = "json";
+            return message;
+        }
+        return new Message(messageName, record.value());
```

Comment:
If I understand correctly, this will fail if `valueSchema != null` and `valueSchema != STRUCT`, i.e. there is a schema set, but it's not a Struct type, such as Array, String, Integer, etc, as we'll still be passing a Connect data type to `ably-java`, which it won't know how to process? 

Have we made a deliberate decision to only support Struct as the top-level schema? If so, we should probably add a more explicit error, else it'll be raised as a bug should somebody try to send an array or similar. I'm not sure it's really necessary to require this? Are not `1`, `"foo"`, `3.145`, `[1, 2, 3]` etc all valid JSON values? The [docs](https://ably.com/docs/api/realtime-sdk/messages#data) suggest we at least support arrays.

If others agree, I'm happy with us just raising a ticket to add support for other top-level types later rather than delaying this PR.

Reply:
@jaley Yes, this is going to only take a Struct as a top level. @QuintinWillison also brought up this question https://github.com/ably/kafka-connect-ably/pull/73#discussion_r881756462 and answered that that was out of the scope of this PR - but looking back I think I might be wrong. My thinking was that there wouldn't be any non-struct top level data with schema, judging from the schemas I tried.  
I now think we need to address this - As we already have our logic that deals with other types in our JSON converter this shouldn't be a big change. WDYT?

Reply:
I think that by far the most common use case will be that the top-level schema is for a Struct, but there's no real need for us to enforce that, as far as I can see. As discussed offline, I think it's fine to just raise a ticket to reconsider this in a separate PR later if we're keen to get support for the majority use case merged ASAP.

This branch, however (line 57), I think will *always* throw an unhelpful exception to the end user? We know that the record has a schema and that schema is not a Struct. The value is therefore a Connect value, which `ably-java` won't know how to handle. Is that correct? If so, and we want to just revisit handling other top-level data types later, I think we should at least change this behaviour so that we throw an exception explicitly describing the problem (unexpected top-level data type, etc), rather than letting `ably-java` throw something more mysterious.

Reply:
That makes sense @jaley I will throw an explicit ConnectException as I do in other places.

Reply:
@jaley I made the change here cf1cdfbafa7b26c32c90e01156dbe5405dfb25b8 - however I realized our previous test were broken - because we already support bytes and also strings. I added support for them in subsequent commits